internal/connector/internal/services/connector_cluster.go

```diff
@@ -505,7 +523,7 @@ func (k *connectorClusterService) UpdateConnectorDeploymentStatus(ctx context.Co
 		return services.HandleGoneError("Connector deployment", "id", deploymentStatus.ID)
 	}
 
-	if err := dbConn.Model(&deploymentStatus).Where("id = ? and version <= ?", deploymentStatus.ID, deploymentStatus.Version).Save(&deploymentStatus).Error; err != nil {
+	if err := dbConn.Where("id = ? and version <= ?", deploymentStatus.ID, deploymentStatus.Version).Save(&deploymentStatus).Error; err != nil {
```

Comment:
This is no doubt affected by https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pull/1637/files#diff-aee57c287964506015af5e0e2839d1e78a19635c9ce232369e8be08a91101d89R508-R517

Reply:
Here I just removed the use of `Model()` with `Save()` as is not supported and can lead to undefined behavior (as per documentation first NOTE: https://gorm.io/docs/update.html)

So I believe we can merge this one and then reason about the whole `// See https://github.com/go-gorm/gorm/issues/6139
	// We can no longer rely on the Save throwing a constraint violation exception when upserting a record that already exists.
	// By first checking that the record existences we ensure the Save only updates an existing record vs trying to create a duplicate.`